core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2662,6 +2662,37 @@ public static void jsonArrayAggAdd(List list, Object element,
     }
   }
 
+  public static String jsonType(Object input) {
+    String result = "";
+    try {
+      Object o = input == null ? null : dejsonize(input.toString());
+      if (o instanceof Integer) {
+        result = "INTEGER";
+      } else if (o instanceof String) {
+        result = "STRING";
+      } else if (o instanceof Float) {
+        result = "FLOAT";
+      } else if (o instanceof Double) {
+        result = "DOUBLE";
+      } else if (o instanceof Long) {
+        result = "LONG";
```

Comment:
Is it really needed to distinguish INTEGER and LONG (So do FLOAT and DOUBLE)? I see MySQL's JSON_TYPE[1] does not provide LONG and DOUBLE output.

[1] https://dev.mysql.com/doc/refman/8.0/en/json-attribute-functions.html#function_json-type

Reply:
A Long can be generated if:
json_type(timestamp '2016-02-24 12:42:25')
The double type contains the float scene so it is converted to a double.
and now Binary types and OPAQUE types: Jackson handle would give you an error.